openssl/src/x509/mod.rs

```diff
@@ -618,6 +620,72 @@ impl Drop for X509Req {
     }
 }
 
+/// A collection of X.509 extensions.
+///
+/// Upholds the invariant that a certificate MUST NOT include more than one
+/// instance of a particular extension, according to RFC 3280 §4.2.
```

Comment:
Could you also note in the comment here that addition order matters, which is why we have this extra logic in the first place?